pkg/aws/ec2/api/helper.go

```diff
@@ -434,77 +445,120 @@ func (h *ec2APIHelper) GetInstanceDetails(instanceId *string) (*ec2.Instance, er
 	return nil, fmt.Errorf("failed to find instance details for input %v", *describeInstanceInput)
 }
 
-// AssignIPv4AddressesAndWaitTillReady assigns IPv4 Address to the interface and waits till the IP Address is attached
-// to the instance
-func (h *ec2APIHelper) AssignIPv4AddressesAndWaitTillReady(eniID string, count int) ([]string, error) {
-	var assignedIPs []string
+func (h *ec2APIHelper) AssignIPv4ResourcesAndWaitTillReady(eniID string, resourceType config.ResourceType, count int) ([]string, error) {
+	var assignedResources []string
+	input := &ec2.AssignPrivateIpAddressesInput{}
 
-	input := &ec2.AssignPrivateIpAddressesInput{
-		NetworkInterfaceId:             &eniID,
-		SecondaryPrivateIpAddressCount: aws.Int64(int64(count)),
+	switch resourceType {
+	case config.ResourceTypeIPv4Address:
+		input = &ec2.AssignPrivateIpAddressesInput{
+			NetworkInterfaceId:             &eniID,
+			SecondaryPrivateIpAddressCount: aws.Int64(int64(count)),
+		}
+	case config.ResourceTypeIPv4Prefix:
+		input = &ec2.AssignPrivateIpAddressesInput{
+			NetworkInterfaceId: &eniID,
+			Ipv4PrefixCount:    aws.Int64(int64(count)),
```

Comment:
I have a question on this field. I saw this comment from AWS SDK comment
```
    // The number of IPv4 prefixes that Amazon Web Services automatically assigns
    // to the network interface. You cannot use this option if you use the Ipv4
    // Prefixes option.
    Ipv4PrefixCount *int64 `type:"integer"`
```
Not sure if this is a concern.

Reply:
In the assign call we make here, we only use `Ipv4PrefixCount` field, and never use `Ipv4Prefixes`, so we are complying with the SDK comment here. Do you have other concern in mind?

Reply:
Got it. Thanks